### ISSUE TITLE
Missing back references

### ISSUE THREAD

**Issue by ashgti** [zeeshanu/learn-regex#40]
Most regular expression engines support back references, its probably worth having a section to discuss how they can be used.

**Comment by mathewreny** [comment]
I agree that a section should be included on backreferences.  "Why you should never use backreferences and avoid all libraries that use them." is my proposed title.

> If all (your)? [a-z]+ were jumping off a bridge would you\\?

Backreferences are not valid regular expressions.  They violate the pumping lemma and therefore remove the main feature of regular expressions -- matching input in linear time.  In fact, backreferences turn an optimal linear problem into an NP-complete mess.  This syntactical caner was introduced by early regex implementers, ignorant of automata theory, to trick others into using their libraries.

I'm passionate about regex theory as you might have guessed.  My angry tone is directed at whoever feature-creeped backreferences into existence, not you.  /rant

**Comment by newmusicdownload** [comment]
So nice.
